rfd/0194-selinux-policy.md

```diff
@@ -0,0 +1,150 @@
+CIL is extremely powerful but much more difficult to understand by humans, it's meant for consumption by SELinux tools.
+
+#### Future updates
+
+The SELinux policy will be updated in the future to support new features and maintain compatibility with existing features.
+Any changes to the SSH service that result in new files being managed, new network activity (listening on new ports, dialing new addresses etc),
+new binaries being executed, or new Linux syscalls being called will require updates to the SELinux policy.
+
+### Installation and updating
+
+The policy source will be embedded in future Teleport agent binaries.
+
+A `selinux` subcommand will be added to Teleport agent binaries to allow users to install and configure the SELinux policy. This subcommand
+will use the Teleport agent configuration file to determine how to configure some of the tunable SELinux policy contexts, such as port
+numbers, file paths, and booleans using the `semanage` tool.
+
+The policy will allow the Teleport agent processes with the `selinux` subcommand only to create or update its SELinux policy. This will be enforced by
+creating a `teleport_selinux_management` domain, which will not be allowed to be a parent of any other Teleport related domain. Otherwise an attacker could
+use the SSH service to run the `selinux` subcommand and potentially modify the policy.
+
+If the SSH service itself was allowed to manage its own SELinux policy, an attacker that compromised the SSH service could modify the policy to
+be more permissive, defeating the point of the policy in the first place.
+
+If SELinux is not present on the host, the subcommand will tell the user this and exit. If SELinux is present but the Teleport agent policy is
+not installed, the subcommand will extract the embedded policy and use `checkmodule`, `semodule_package`, and `semodule` to compile and install the policy.
+
+Installation scripts and the `teleport-update` tool will be updated to install and configure the SELinux policy with the `selinux` subcommand.
+
+SELinux has 3 modes, disabled, permissive (policy violations are only logged) and enforcing (attempts to violate the policy are denied).
+The SELinux mode will not be modified by the `selinux` subcommand, users will be responsible for enabling SELinux if they so choose.
+
+The `selinux` subcommand will have a `--dry-run` flag that will print what actions it would need to take to manage the policy if run normally and
+print the policy that would be installed based on the provided Teleport agent configuration file.
+
+### Compliance
+
+To ensure the SSH agent SELinux policy is installed and SELinux is configured to enforce it, a `--selinux` flag will be added to the `teleport start` subcommand
+of Teleport agents. If the SELinux policy is not installed or SELinux is not present and in enforcing mode and `--selinux` is passed, the Teleport agent
```

Comment:
Need a separate section explaining how we will interact with existing SELinux policies that may already be in place. In that case I think we should not override and users should be able to opt-out of our builtin one.

Reply:
Added a note on this. I don't think it's feasible to inspect every installed policy to see if they interact with Teleport at all, so we'll just have to do a best effort check.